korman/exporter/etlight.py

```diff
@@ -384,25 +386,25 @@ def _prep_for_lightmap(self, bo, toggle):
             uv_textures.active = uvtex
 
             # if the artist hid any UVs, they will not be baked to... fix this now
-            bpy.ops.object.mode_set(mode="EDIT")
-            bpy.ops.uv.reveal()
-            bpy.ops.object.mode_set(mode="OBJECT")
+            with self._set_mode("EDIT"):
+                bpy.ops.uv.reveal()
             self._associate_image_with_uvtex(uv_textures.active, im)
-            bpy.ops.object.mode_set(mode="EDIT")
 
-            # prep the uvtex for lightmapping
-            bpy.ops.mesh.select_all(action="SELECT")
-            bpy.ops.uv.average_islands_scale()
-            bpy.ops.uv.pack_islands()
+            # Meshes with modifiers need to have islands packed to prevent generated vertices
+            # from sharing UVs. Sigh.
+            if self.is_collapsed(bo):
+                self._report.warn("'{}': packing islands in UV Texture '{}' due to modifier collapse", bo.name, uv_base.name)
+                with self._set_mode("EDIT"):
+                    bpy.ops.mesh.select_all(action="SELECT")
+                    bpy.ops.uv.pack_islands()
         else:
             # same thread, see Sirius's suggestion RE smart unwrap. this seems to yield good
             # results in my tests. it will be good enough for quick exports.
             uvtex = uv_textures.new(self.lightmap_uvtex_name)
```

Comment:
```suggestion
            uvtex = uv_textures.new(self.lightmap_uvtex_name)
            uv_textures.active = uvtex
```
So apparently we never caught on to this small oversight with horrible consequences :rofl: We need to set the UVtex as active, otherwise Bad Things happen (like messing up the user's UVs, baking useless lightmaps, and increasing baking times tenfold).